### PR TITLE
Feature/redux instructor stats

### PR DESCRIPTION
**Pull request by am-stewart**
## Description

This pull request focuses on the instructor's stats being displayed on their dashboard when logged in. We connected the dashboard to Redux and now have the information connected to the backend rather than pulling information from the dummy data.

Collaborators: 
[Jorge Evangelista](https://github.com/JorgeEvangelista500)
Fixes # (issue)

## Loom Video

https://www.loom.com/share/de9712776add443590f5cb3f7beca40c

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by bugsbrog — APPROVED** [review]
Love that you used props and mapStateToProps in InstructorStats.js. Also, love how you cleaned up your code and changed the a href tags to links in InstructorSidebar.js. I like how you added dummy data (I think it's a lot easier to work with when there is actual data). Love how you commented out the parent2 route path in index.js because it is a duplicate. I'd say to get rid of commented-out code, but I think Jessica might've gotten rid of it. LOVE LOVE LOVE how you renamed the export const names so they matched in instructorActions.js.

**Review by mnenslei — APPROVED** [review]
The code looks great, I liked how you switched the sidebar a href tags to links and made it uniform throughout the page, also the redux implementation looks good too. Great work!
There is some slight repetition on the Dashboard/InstructorStat.js page with the cards that could possibly be made into a separate reusable component and save space on this page's component. Other than that this looks great, amazing work Allison and Jorge!

**Review by cavazosgeorge — APPROVED** [review]
I like the naming conventions you guys used in declaring your variables. Easy to understand what everything is doing. Great job!

File: src/components/pages/Dashboard/InstructorStats.js [hunk 1]
```diff
@@ -1,70 +1,115 @@
...
+function InstructorStats(props) {
+  const { authState, authService } = useOktaAuth();
   const { idToken } = authState;
+  const dispatch = useDispatch();
   const [stats, setStats] = useState(initialValues);
 
   useEffect(() => {
-    // setStats(getStats()); // ENABLE WHEN READY TO CONNECT STATS TO STATE
-  }, []);
+    if (!props.user.name) {
```
Comment (JessWillCode) [inline src/components/pages/Dashboard/InstructorStats.js:30]: Great job incorporating the scenario of no user!

File: src/components/pages/Dashboard/InstructorStats.js [hunk 2]
```diff
@@ -1,70 +1,115 @@
...
-              <div class="stat-content">
-                <p>{stat.title}</p>
-                <h3>{stat.value}</h3>
-              </div>
-            </Card>
-          );
-        })}
+        <Card id="ant-card-stat">
+          <div class="stat-icon-wrapper">
+            <UserOutlined style={{ fontSize: '40px', color: '#F79E1B' }} />
```
Comment (JessWillCode) [inline src/components/pages/Dashboard/InstructorStats.js:70]: Is there a way to not have inline style? I know that Ant design has been giving us trouble, so no worries if not!

File: src/components/pages/InstructorHome/InstructorSidebar.js [hunk 3]
```diff
@@ -35,23 +35,23 @@ function InstructorSidebar() {
...
         </Menu.Item>
 
         <Menu.Item key="2" icon={<HomeOutlined />}>
-          <a href="/instructor">Overview</a>
+          <Link to="/instructor">Overview</Link>
         </Menu.Item>
 
         <Menu.Item key="3" icon={<ReadOutlined />}>
-          <a href="/instructor-all-classes">Courses</a>
+          <Link to="/instructor-all-classes">Courses</Link>
```
Comment (JessWillCode) [inline src/components/pages/InstructorHome/InstructorSidebar.js:46]: Great job changing these to links. This makes the routes consistent!

File: src/index.js [hunk 4]
```diff
@@ -117,7 +117,7 @@ function App() {
           <Route path="/implicit/callback" component={LoginCallback} />
           <Route path="/instructor" component={InstructorHome} />
           <Route path="/parent" component={ParentDashboard} />
-          <Route path="/parent2" component={ParentHome} />
+          {/* <Route path="/parent2" component={ParentHome} /> */}
```
Comment (JessWillCode) [inline src/index.js:120]: Great catch finding this issue!

File: src/redux/actions/instructorActions.js [hunk 5]
```diff
@@ -35,12 +36,14 @@ export const getusers = () => async dispatch => {
...
+    const res = await axiosWithAuth(idToken).get(`/instructor/courses`);
+    const courses = res.data.map(course => {
+      const now = Date.now();
+      return { ...course, active: now < new Date(course.end_date).getTime() };
+    });
     dispatch({
-      type: GET_INSTRUCTOR_CLASSES,
-      payload: res.data,
+      type: GET_INSTRUCTOR_COURSES,
+      payload: courses,
```
Comment (JessWillCode) [inline src/redux/actions/instructorActions.js:46]: I appreciate that you kept the names consistent throughout!

**Review by JessWillCode — APPROVED** [review]
Thank you for the clean code, mindful changes, and consistent spacing. The code was super easy to read, and I can tell you really thought about the state changes you wanted to implement.

**Pull request merged**